pallets/pallet-asset-switch/src/xcm/match/mod.rs

```diff
@@ -55,14 +60,21 @@ where
 			XcmExecutorError::AssetNotHandled
 		})?;
 		ensure!(id == a.id, XcmExecutorError::AssetNotHandled);
+		// 3. Verify the stored asset is a fungible one.
+		let Fungibility::Fungible(_) = fun else {
+			log::info!(target: LOG_TARGET, "Stored remote fee asset {:?} is not a fungible one.", remote_xcm_fee);
+			return Err(XcmExecutorError::AssetNotHandled);
+		};
```

Comment:
The implemented trait is `MatchesFungible`, so it's not supposed to work with non-fungible assets. If a non-fungible asset is set as XCM fee, there would need to be a different XCM component, which we don't provide here.

Reply:
Okay. I am asking, because I was considering adding those checks in the extrinsic. What you think? Would it unnecessarily restrict the pallet too much?

Reply:
Yeah I would not put it in now.